graf3d/eve7/src/REveBoxSet.cxx

```diff
@@ -357,20 +392,53 @@ void REveBoxSet::BuildRenderData()
 
             if (fValueIsColor) {
                fRenderData->PushI(int(b->fValue));
-            } else if (fSingleColor == false) {
+            } else {
                UChar_t c[4] = {0, 0, 0, 0};
                fPalette->ColorFromValue(b->fValue, fDefaultValue, c);
 
                int value = c[0] + c[1] * 256 + c[2] * 256 * 256;
-
                // printf("box val [%d] values (%d, %d, %d) -> int <%d>\n", b.fValue, c[0], c[1], c[2],  value);
                fRenderData->PushI(value);
             }
          }
       }
    }
+   if (Instanced()) {
+      // printf(" >>> resize render data %d \n", 4 * fTexX * fTexY);
+      fRenderData->ResizeV(4 * fTexX * fTexY);
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get int value for color. Used for case of instancing.
+///
+unsigned int REveBoxSet::GetColorFromDigit(REveDigitSet::DigitBase_t &digi)
+{
+   if (fSingleColor == false) {
+      if (fValueIsColor) {
+        UChar_t* c = (UChar_t*) & digi.fValue;
+        return (c[0] << 16) + (c[1] << 8) + c[2];
+      } else {
+         // printf("palette\n");
+         UChar_t c[4] = {0, 0, 0, 0};
+         fPalette->ColorFromValue(digi.fValue, fDefaultValue, c);
+         return (c[0] << 16) + (c[1] << 8) + c[2];
+      }
+   }
+   // printf("main color %d\n", GetMainColor());
+   UChar_t c[4] = {0, 0, 0, 0};
+   REveUtil::ColorFromIdx(GetMainColor(), c);
+   // printf("rgb %d %d %d\n", c[0], c[1], c[2]);
+   return (c[0] << 16) + (c[1] << 8) + c[2]; // AMT perhaps this can be ignored
+}
+
+float REveBoxSet::GetColorFromDigitAsFloat(REveDigitSet::DigitBase_t &digit)
+{
+   unsigned int c = GetColorFromDigit(digit);
+   return *reinterpret_cast<float*>(&c);
```

Comment:
Please fix warning:
```
/graf3d/eve7/src/REveBoxSet.cxx:438:12: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
```
Just use place C cast here:
```
uint32_t c = GetColorFromDigit(digit);
return *((float *)&c);
```
Is it really float coded as binary in integer?

Reply:
@linev Thank you for the suggestion. The change is included in the branch.